### ISSUE TITLE
【CVPR'19】Quasi-unsupervised Color Constancy

### ISSUE THREAD

**Issue by huuuuusy** [huuuuusy/CV-Learning-Everyday#2]
# Link
[GitHub](https://github.com/claudio-unipv/quasi-unsupervised-cc)
[Paper](http://openaccess.thecvf.com/content_CVPR_2019/papers/Bianco_Quasi-Unsupervised_Color_Constancy_CVPR_2019_paper.pdf)

**Comment by huuuuusy** [comment]
# Abstract
We present here a method for computational color constancy in which a deep convolutional neural network is trained to detect achromatic pixels in color images after they have been converted to grayscale. The method does not require any information about the illuminant in the scene and relies on the weak assumption, fulfilled by almost all images available on the Web, that training images have been approximately balanced. Because of this requirement we define our method as quasi-unsupervised. After training, unbalanced images can be processed thanks to the preliminary onversion to grayscale of the input to the neural network. The results of an extensive experimentation demonstrate that the proposed method is able to outperform the other unsupervised methods in the state of the art being, at the same time, flexible enough to be supervisedly fine-tuned to reach performance comparable with those of the best supervised methods.

**Comment by huuuuusy** [comment]
# System 

![image](https://user-images.githubusercontent.com/32117906/62818184-60700580-bb76-11e9-8975-4bbf1785818b.png)

## End to end quasi-unsupervised training
• the illuminant is estimated as the normalized weighted sum of the input pixels plus some Gaussian noise n (to favor large weights)
• the parameters of the CNN are optimized to minimize the achromatic loss without any ground truth illuminant

**Comment by huuuuusy** [comment]
# Schematic View

![image](https://user-images.githubusercontent.com/32117906/62818209-f9068580-bb76-11e9-9438-2af8b0250d17.png)

**Comment by huuuuusy** [comment]
# Network Architecture

![image](https://user-images.githubusercontent.com/32117906/62818227-34a14f80-bb77-11e9-9010-ac36dd257f37.png)

**Comment by huuuuusy** [comment]
# Learning Setting

### Quasi-unsupervised  Learning --> No color constancy dataset

**Ilsvrc12**

1. Publicly available for the Internet Large Scale Visual Recognition Challenge which probably represents the most popular benchmark for image recognition
2. Consists of approximately 1.2 millions samples taken from of 1000 different categories

**Places365**

1. Serve as benchmark for scene recognition systems
2. Includes about 1.8 millions images representing 365 different categories of scenes

**Flickr100k**

1. Evaluate image retrieval algorithms
2. Consists of 100071 images collected from the Flickr photo sharing service by searching for the 146 most popular tags

![Screenshot from 2019-08-10 14-04-05](https://user-images.githubusercontent.com/32117906/62818268-cdd06600-bb77-11e9-9678-e62db98e0d36.png)

**Comment by huuuuusy** [comment]
# Model Variants

## Variants
1. Differ in the data set used for training 

- Ilsvrc12
- Places365 
- Flickr100k

2. Differ in the type of input 

- Equalized grayscale 
- Gradient directions
- Both

3. Differ in the learning setting 

- Quasi-unsupervised
- Supervised fine tuning

## Main differences in the behavior of the variants

1. The networks **processing only grayscale images select pixels in regions that are easier to recognize** with respect to those obtained using also gradient directions
2. **Fine tuned networks tend to select less pixels than the others.** This may depend on the fact that, during the fine tuning, we removed the noise term that pushed the network to output large weights
3. **The output of the networks only marginally depend on the training set used.** In particular the weights found by networks trained on Ilsvrc12 and Places365 are, most of the times, very similar

**Comment by huuuuusy** [comment]
# Visualization

## Color Checker images processed by variants of the method
![image](https://user-images.githubusercontent.com/32117906/62818401-7e8b3500-bb79-11e9-8db4-adbc28a69962.png)

## NUS images processed by variants of the method
![image](https://user-images.githubusercontent.com/32117906/62818405-864ad980-bb79-11e9-9780-0f075d49ffec.png)

**Comment by huuuuusy** [comment]
# Evaluation
![image](https://user-images.githubusercontent.com/32117906/62818433-bd20ef80-bb79-11e9-92e1-7c59745edbf1.png)

![image](https://user-images.githubusercontent.com/32117906/62818441-df1a7200-bb79-11e9-9b03-ae8b2bd4c233.png)
**in dataset(in-db):**
the method is trained/ tuned with cross validation on the same color constancy dataset on which is tested
**cross dataset(cross-db):**
the method is trained/ tuned on one color constancy dataset and tested on a different one
**no dataset(no-db):**
the method is not trained/ tuned on any color constancy dataset

![image](https://user-images.githubusercontent.com/32117906/62818453-f6595f80-bb79-11e9-9549-c732ce846a18.png)
**The proposed method**
• Outperforms all the purely unsupervised algorithms by a large margin
• Outperforms all the supervised methods in the cross-dataset setting
• After fine tuning (FT), it rivals with the best supervised methods (best mean error and second best median on NUS)

**Comment by huuuuusy** [comment]
# Test by myself
**pytorch-gpu + python3.7**
![image](https://user-images.githubusercontent.com/32117906/62818562-69170a80-bb7b-11e9-9c1c-8fefbf33f6e3.png)
![image](https://user-images.githubusercontent.com/32117906/62818563-7af8ad80-bb7b-11e9-84bb-b67f0de12d40.png)
![image](https://user-images.githubusercontent.com/32117906/62818566-88ae3300-bb7b-11e9-8fcc-3f653a86c826.png)
![image](https://user-images.githubusercontent.com/32117906/62818569-95328b80-bb7b-11e9-8d54-1e13e82e1a31.png)

**Comment by huuuuusy** [comment]
> # Test by myself
> **pytorch-gpu + python3.7**
> ![image](https://user-images.githubusercontent.com/32117906/62818562-69170a80-bb7b-11e9-9c1c-8fefbf33f6e3.png)
> ![image](https://user-images.githubusercontent.com/32117906/62818563-7af8ad80-bb7b-11e9-84bb-b67f0de12d40.png)
> ![image](https://user-images.githubusercontent.com/32117906/62818566-88ae3300-bb7b-11e9-8fcc-3f653a86c826.png)
> ![image](https://user-images.githubusercontent.com/32117906/62818569-95328b80-bb7b-11e9-8d54-1e13e82e1a31.png)

This part needs a pre-processing stage before color constancy, however the results of pre-processed images are too bright to be used. I do not find the reason of this bug. The pictures above are no pre-processing results.
